validator/db/kv/attester_protection.go

```diff
@@ -258,13 +309,25 @@ func (s *Store) batchAttestationWrites(ctx context.Context) {
 // and resets the list of batched attestations for future writes.
 // This function notifies all subscribers for flushed attestations
 // of the result of the save operation.
-func (s *Store) flushAttestationRecords(ctx context.Context) {
+func (s *Store) flushAttestationRecords(ctx context.Context, records []*AttestationRecord) {
+	if s.batchedAttestationsFlushInProgress.IsSet() {
+		log.Error("Attempted to flush attestation records when already in progress")
```

Comment:
Should this be at error level? maybe debug is appropriate. We should only have error logs when they are actionable to the user, and this is opaque from a user's perspective

Reply:
This should never ever happen. If it is happening, then it's a development bug that was recently introduced. I will add a comment to reflect the significance